tests/data/generator/two_party_negotiation/message.py

```diff
@@ -0,0 +1,308 @@
+        assert self.is_set("conditions"), "'conditions' content is not set."
+        return cast(
+            Optional[
+                Union[str, Dict[str, int], FrozenSet[DataModel], Dict[str, float]]
+            ],
+            self.get("conditions"),
+        )
+
+    @property
+    def description(self) -> str:
+        """Get the 'description' content from the message."""
+        assert self.is_set("description"), "'description' content is not set."
+        return cast(str, self.get("description"))
+
+    @property
+    def flag(self) -> bool:
+        """Get the 'flag' content from the message."""
+        assert self.is_set("flag"), "'flag' content is not set."
+        return cast(bool, self.get("flag"))
+
+    @property
+    def items(self) -> Tuple[Unit, ...]:
+        """Get the 'items' content from the message."""
+        assert self.is_set("items"), "'items' content is not set."
+        return cast(Tuple[Unit, ...], self.get("items"))
+
+    @property
+    def number(self) -> int:
+        """Get the 'number' content from the message."""
+        assert self.is_set("number"), "'number' content is not set."
+        return cast(int, self.get("number"))
+
+    @property
+    def price(self) -> float:
+        """Get the 'price' content from the message."""
+        assert self.is_set("price"), "'price' content is not set."
+        return cast(float, self.get("price"))
+
+    @property
+    def proposal(self) -> Optional[Dict[str, IOTApp7]]:
+        """Get the 'proposal' content from the message."""
+        assert self.is_set("proposal"), "'proposal' content is not set."
+        return cast(Optional[Dict[str, IOTApp7]], self.get("proposal"))
+
+    @property
+    def query(self) -> DataModel:
+        """Get the 'query' content from the message."""
+        assert self.is_set("query"), "'query' content is not set."
+        return cast(DataModel, self.get("query"))
+
+    @property
+    def rounds(self) -> FrozenSet[int]:
+        """Get the 'rounds' content from the message."""
+        assert self.is_set("rounds"), "'rounds' content is not set."
+        return cast(FrozenSet[int], self.get("rounds"))
+
+    def _is_consistent(self) -> bool:
+        """Check that the message follows the two_party_negotiation protocol."""
+        try:
+            assert (
+                type(self.dialogue_reference) == tuple
+            ), "dialogue_reference must be 'tuple' but it is not."
+            assert (
+                type(self.dialogue_reference[0]) == str
+            ), "The first element of dialogue_reference must be 'str' but it is not."
+            assert (
+                type(self.dialogue_reference[1]) == str
+            ), "The second element of dialogue_reference must be 'str' but it is not."
+            assert type(self.message_id) == int, "message_id is not int"
+            assert type(self.target) == int, "target is not int"
+
+            # Light Protocol 2
```

Comment:
Maybe instead say `Light Protocol Rule 2` so it is more explicit?

Reply:
same below

Reply:
Done